dev/preinstalled_providers.json

```diff
@@ -0,0 +1,9 @@
+[
+    "http",
+    "common.io",
+    "common.sql",
+    "ftp",
+    "http",
+    "imap",
+    "sqlite"
```

Comment:
Other question anyway, just thinking of beautification - why adding a new JSON file? Have you considered using a YAML instead?

Reply:
Because this file needs to be (ideally) read withoug installing any extra dependency. YAML requires installing pyyaml and since we want to read it in the "build environment" - which is created by adding hatchling and gitpython (and other transitive dependencies). If we add pyyaml to it, we need to install it (and all transitive dependencies every time `pip install` executes for Airflow. We do not want it.

I actually considered (and maybe that's what I will do eventually) to use plain text file - same as we already do for installed_providers.txt - this is a file that contains list of providers that are installed in the production image by default. I actually think I will eventually do that, I just have to rename the original file (installed_providers.txt -> prod_image_providers.txt as there will be too big of an ambiguity).

Reply:
Yeah. Just following what was already done with `installed_providers.txt` is much better. I even had pre-commit that was sorting them :) that required just a small refactor to sort both. I also added comments capability to both of them to explain what they are for - now when there are two, better to explain more precisely what they are about.

Reply:
And moved both files to `dev` where they belong.